tests/functional/test_benchmark_2d_case_3.py

```diff
@@ -0,0 +1,189 @@
+"""
+Module containing tests for cases 3a and 3b from the 2d benchmark study [1].
+
+The tests check whether effective permeabilities and boundary conditions are correctly
+assigned. Since we solved the actual model, we also implicitly check that the model does
+not crash. However, we are currently not checking that the solution is the "correct"
+one.
```

Comment:
Is it correct to say that these tests (and those for 3d case 3) verifies that the benchmark setup and the tests are consistently set up, but not that the values are actually the right ones (for that, the flow problem must be solved, errors computed etc. etc.)? If so, these tests will fail:
* If someone changes the setup, but not the test;
* If we change the way a single phase fluid model is set up, for instance by changing method names for specifying boundary conditions, but do not change the test;
* if we introduce an error in the permeability scaling with specific volume - this is the only non-trivial quantitative test that I can see here.
I also realize that running the tutorials check will verify that the 2d models also run, though not the 3d case.

I see four alternatives (or two pairs of alternatives) with regards to testing:
1. Leave the tests as they are, but introduce some verification that the 3d case actually runs. If the 3d case is slow, this can be a skipped test that only runs once a week.
2. Same as 1., but also include some weak check that detects changes in the solution (in the spirit of functional tests), say, in the form of total flux through the in- or outlet boundaries.
3. Introduce tests similar to those in this file, also for 2d case 1 and 3d case 3.
4. Drop the tests of the models, except verification that they run, with the motivation that the tests add more entropy and possibly maintenance than they enhance security.
Of course, both 3. and 4. can in a sense be combined with 1. or 2.

A verification that 3d case 3 does not crash should be made before this PR is merged. My preference would be to implement 2., but I think we can postpone that somewhat, and possibly reassign the task. With regards to the present file, I lean towards 4., but I am open to other opinions - let me know.

Reply:
Alternative 2 could be accommodated in the tutorial (with an assert on the solution, say).

Reply:
If I may give my two cents on this...

I believe a decent compromise between maintenance and having some level of confidence in the benchmarks is testing the boundary conditions and permeability specifications. These are, essentially, the two aspects that are crucial to get it right when it comes to the flow benchmarks (in addition to the geometry that is). At the end of the day, the incompressible flow model I'd say is quite well tested via funcional and integration tests.

The next level of rigour, which is checking the solution, is indeed non-trivial, since this will probably be quite sensitive to the mesh that `gmsh` produces (this is why we suggest to compare order of convergence in the rigorous paper for system tests). Self-convergence could be an option, but it will require some effort in terms of implementation.

Reply:
These are valid points, @jhabriel. Let's leave the tests as they are.